From 90251ef253564e7e53c754205cb506e863a93a6c Mon Sep 17 00:00:00 2001
From: Sreerenj Balachandran <sreerenj.balachandran@intel.com>
Date: Thu, 26 Nov 2015 19:13:43 +0200
Subject: [PATCH] videoparsers: h264: Disable passthorugh mode enabling

Enabling passthorugh mode for optimization is cauing multiple
issues while parsing MVC streams. Specifically if streams have
two separate layers (base-view and non-base-view).

Proper fixes needed in many places:
(handle prefix nal unit, handle non-base-view slice nal extension,
fix the picture_start detection for multi-layer-mvc streams etc)

https://bugzilla.gnome.org/show_bug.cgi?id=758656
---
 gst/vaapi/gsth264parse.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gst/vaapi/gsth264parse.c b/gst/vaapi/gsth264parse.c
index 915c2d7..445a791 100644
--- a/gst/vaapi/gsth264parse.c
+++ b/gst/vaapi/gsth264parse.c
@@ -2356,6 +2356,9 @@ gst_h264_parse_pre_push_frame (GstBaseParse * parse, GstBaseParseFrame * frame)
     }
   }

+  /* Fixme: setting passthrough mode for MVC streams causing multiple
+   * issues. Disabing passthourgh mode for now */
+#if 0
   /* If SPS/PPS and a keyframe have been parsed, and we're not converting,
    * we might switch to passthrough mode now on the basis that we've seen
    * the SEI packets and know optional caps params (such as multiview).
@@ -2367,6 +2370,7 @@ gst_h264_parse_pre_push_frame (GstBaseParse * parse, GstBaseParseFrame * frame)
       gst_base_parse_set_passthrough (parse, TRUE);
     }
   }
+#endif

   gst_h264_parse_reset_frame (h264parse);

--
2.5.0

